src/pages/logs-detail-page.tsx

```diff
@@ -71,6 +71,10 @@ const LogsDetailPage: React.FunctionComponent = () => {
     runQuery({ severityValue: severityFilterValue });
   };
 
+  const handleQueryChange = (newQuery: string | undefined) => {
+    setQuery(newQuery ?? '');
+  };
```

Comment:
Could we instead define `LogsToolbar`'s `onQueryChange` and `LogsQueryInput`'s `onChange` so that they only accept a string (not `undefined`)? Then it looks like these `handleQueryChange` functions would not be needed.